packages/pluggableWidgets/line-chart-web/tests/e2e/specs/LineChart.spec.ts

```diff
@@ -0,0 +1,177 @@
+            const screenshotElem = $(".mx-name-containerBoth");
+            screenshotElem.waitForDisplayed({ timeout: 5000 });
+            screenshotElem.scrollIntoView();
+
+            browser.saveElement(screenshotElem, "lineChartGridLinesBoth");
+            expect(browser.checkElement(screenshotElem, "lineChartGridLinesBoth")).toEqual(0);
+        });
+    });
+
+    describe("legend", () => {
+        it("renders with legend and compares with a screenshot baseline", () => {
+            browser.setWindowRect(0, 0, 1200, 900);
+            const screenshotElem = $(".mx-name-containerLegend");
+            screenshotElem.waitForDisplayed({ timeout: 5000 });
+            screenshotElem.scrollIntoView();
+
+            browser.saveElement(screenshotElem, "lineChartLegend");
+            expect(browser.checkElement(screenshotElem, "lineChartLegend")).toEqual(0);
+        });
+
+        it("renders without legend and compares with a screenshot baseline", () => {
+            browser.setWindowRect(0, 0, 1200, 900);
+            const screenshotElem = $(".mx-name-containerNoLegend");
+            screenshotElem.waitForDisplayed({ timeout: 5000 });
+            screenshotElem.scrollIntoView();
+
+            browser.saveElement(screenshotElem, "lineChartNoLegend");
+            expect(browser.checkElement(screenshotElem, "lineChartNoLegend")).toEqual(0);
+        });
+    });
+
+    describe("axis label", () => {
+        it("renders x axis label and compares with a screenshot baseline", () => {
+            browser.setWindowRect(0, 0, 1200, 900);
+            const screenshotElem = $(".mx-name-containerXLabel");
+            screenshotElem.waitForDisplayed({ timeout: 5000 });
+            screenshotElem.scrollIntoView();
+
+            browser.saveElement(screenshotElem, "lineChartXLabel");
+            expect(browser.checkElement(screenshotElem, "lineChartXLabel")).toEqual(0);
+        });
+
+        it("renders y axis label legend and compares with a screenshot baseline", () => {
+            browser.setWindowRect(0, 0, 1200, 900);
+            const screenshotElem = $(".mx-name-containerYLabel");
+            screenshotElem.waitForDisplayed({ timeout: 5000 });
+            screenshotElem.scrollIntoView();
+
+            browser.saveElement(screenshotElem, "lineChartYLabel");
+            expect(browser.checkElement(screenshotElem, "lineChartYLabel")).toEqual(0);
+        });
+
+        it("renders x+y axis label legend and compares with a screenshot baseline", () => {
+            browser.setWindowRect(0, 0, 1200, 900);
+            const screenshotElem = $(".mx-name-containerXYLabels");
+            screenshotElem.waitForDisplayed({ timeout: 5000 });
+            screenshotElem.scrollIntoView();
+
+            browser.saveElement(screenshotElem, "lineChartXYLabels");
+            expect(browser.checkElement(screenshotElem, "lineChartXYLabels")).toEqual(0);
+        });
+    });
+
+    describe("dimensions", () => {
```

Comment:
If we're going over the widget properties like this, maybe it makes more sense to include all 6 options in the specs?
so:
- width = percentage + height = percentage of width
- width = percentage + height = pixels
- width = percentage + height = percentage of parent
- width = pixels + height = percentage of width
- width = pixels + height = pixels
- width = pixels + height = percentage of parent

Similarly to what I did in the test project. You'll have to include a container with a fixed height in the ones that have a height based on the parent, you can also find that in the test project.

Reply:
Done! 😀